### ISSUE TITLE
Why isn't there Equal and NotEqual numeric types?

### ISSUE THREAD

**Issue by JohnReedLOL** [fthomas/refined#173]
Equal numeric type...
```scala
scala> val a: Int Refined Equal[W.`5`.T] = 5
a: Int Refined Equal[Int(5)] = 5
```

Not equal numeric type...
```scala
scala> val b: Int Refined NotEqual[W.`5`.T] = 6
b: Int Refined NotEqual[Int(5)] = 6
```

**Comment by fthomas** [comment]
@JohnReedLOL There is an [`Equal`](https://github.com/fthomas/refined/blob/10e7709d239b00001ed06961595a95405c03c770/core/shared/src/main/scala/eu/timepit/refined/generic.scala#L18) predicate in the `generic` module that also can be used for numeric types:
```scala
import eu.timepit.refined.W
import eu.timepit.refined.auto._
import eu.timepit.refined.api.Refined
import eu.timepit.refined.boolean.Not
import eu.timepit.refined.generic.Equal

scala> val a: Int Refined Equal[W.`5`.T] = 5
a: Int Refined Equal[Int(5)] = 5

scala> val b: Int Refined Not[Equal[W.`5`.T]] = 6
b: Int Refined Not[Equal[Int(5)]] = 6
```
It is in `generic` because it can also be used for `String` and `Boolean`. Would you mind changing #174 to just a comment in `numeric.scala` that `Equal` from `generic` can also be used for numeric types?

**Comment by JohnReedLOL** [comment]
Done. Thanks

**Comment by fthomas** [comment]
"Fixed by" #174. Thanks again.
